### DISCUSSION TITLE
Not possible to use a global 'allure' variable of karma-allure-reporter in angular 17 in the tests

### DISCUSSION THREAD

**Discussion by mourav2222** [allure-framework/allure2#2575] [Questions & Support] [answered]
Hi,

i configured angular 17 with karma-allure-reporter and the allure-results with tests were generated.
We use angular 17 with test runner framework karma (not jest) and jasmine as test framework in our project, because i ask about workaround.
I want to use the Step and Description feature of allure report.
But i can't use **allure** global variable of karma-allure-reporter in tests. Do someone have an idea, how can you solve this issue? 
For allure integration with karma in angular 17 i used this example.

Not helped any of this attempt to import **allure** variable in test.

```
//const allure = require("karma-allure-reporter")
//import { allure } from '../../karma.conf.js';
// const allure = require('../../karma.conf.js')

```
https://github.com/allure-examples/allure-karma-example

![image](https://github.com/allure-framework/allure2/assets/76436807/9022fcf9-60bd-404a-8cce-b3aa98a240e1)

![image](https://github.com/allure-framework/allure2/assets/76436807/31709953-5569-4378-9203-2115beb5445e)

![image](https://github.com/allure-framework/allure2/assets/76436807/8a14ffad-ea38-49d4-bcc9-21033e70427f)

Best regards

**Comment by baev** [comment, marked as answer]
[karma is deprecated](https://github.com/karma-runner/karma?tab=readme-ov-file#karma-is-deprecated-and-is-not-accepting-new-features-or-general-bug-fixes), so we are dropping support for Allure Karma as well. Please consider using one of the modern test frameworks if possible
